### PR TITLE
python3Packages.hug: disable broken marshmallow tests

### PR DESCRIPTION
**Pull request by urbas**
###### Motivation for this change

This change fixes [this build failure](https://hydra.nixos.org/build/134827324).

###### Things done

Built the package on NixOS aarch64 (in addition to x86-64 Linux).

Output of `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`:
```
3 packages updated:
python3.7-hug python3.8-hug python3.9-hug

3 packages built:
python37Packages.hug python38Packages.hug python39Packages.hug

```

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/hug/default.nix [hunk 1]
```diff
@@ -20,19 +20,29 @@ buildPythonPackage rec {
...
 
-  nativeBuildInputs = [ pytestrunner ];
   propagatedBuildInputs = [ falcon requests ];
 
-  checkInputs = [ mock marshmallow pytest numpy ];
-  checkPhase = ''
-    mv hug hug.hidden
-    # some tests attempt network access
-    PATH=$out/bin:$PATH pytest -k "not (test_request or test_datagram_request)"
+  checkInputs = [ mock marshmallow pytestCheckHook pytestrunner numpy ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/hug/default.nix:25]: Isn't pytestrunner deprecated?

File: pkgs/development/python-modules/hug/default.nix [hunk 2]
```diff
@@ -20,19 +20,29 @@ buildPythonPackage rec {
...
+
+  preCheck = ''
+    # some tests need the `hug` CLI on the PATH
+    PATH=$out/bin:$PATH
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/hug/default.nix:29]:
```suggestion
    export PATH=$out/bin:$PATH
```

File: pkgs/development/python-modules/hug/default.nix (hunk 1 above)
Comment (urbas) [reply pkgs/development/python-modules/hug/default.nix:25]: I tried to remove it and the build stopped working with this error during `setuptoolsBuildPhase`:
```
ERROR: Could not find a version that satisfies the requirement pytest-runner
ERROR: No matching distribution found for pytest-runner
```

Comment (SuperSandro2000) [reply pkgs/development/python-modules/hug/default.nix:25]: I think in other packages we usually remove that dependency.

Comment (urbas) [reply pkgs/development/python-modules/hug/default.nix:25]: With patching? Could you point me to an example?

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 109615` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>python37Packages.hug</li>
<li>python38Packages.hug</li>
<li>python39Packages.hug</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 109615` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>3 packages failed to build and are new build failures:</summary>
  <ul>
<li>python37Packages.hug: log <a href="https://termbin.com/f6uw">https://termbin.com/f6uw</a></li>
<li>python38Packages.hug: log <a href="https://termbin.com/5qpe">https://termbin.com/5qpe</a></li>
<li>python39Packages.hug: log <a href="https://termbin.com/i1su">https://termbin.com/i1su</a></li>
  </ul>
</details>

File: pkgs/development/python-modules/hug/default.nix (hunk 1 above)
Comment (urbas) [reply pkgs/development/python-modules/hug/default.nix]: @SuperSandro2000:  I added a `postPatch` hook that removes the pytest-runner dependency. PTAL

**Comment by SuperSandro2000** [comment]
Next time please do not add fixup commits because github does not fixup them correctly.

**Comment by urbas** [comment]
> Next time please do not add fixup commits because github does not fixup them correctly.

Apologies for that. I added the `fixup!` prefix more to give an indication that I will rebase before the PR gets merged. I wasn't sure if the commit was good or not so I wanted to keep it separate (so it can be easily reviewed in isolation).

I will from now on force-push all follow-up PR changes into a single commit (squashed) unless told otherwise.

**Pull request merged**